test/test/erc721app.js

```diff
@@ -0,0 +1,73 @@
+const BigNumber = require('bignumber.js');
+
+const { expect } = require("chai")
+  .use(require("chai-as-promised"))
+  .use(require("chai-bignumber")(BigNumber))
+
+const { TestToken721Address, polkadotRecipientSS58, polkadotRecipient, bootstrap } = require('../src/fixtures');
+
+const { ChannelId } = require("../src/helpers");
+
+describe('Bridge', function () {
+  const ethTokenContract = TestToken721Address;
+  const ethTokenId = Date.now();
+  let ethOwner;
+  let ethUser;
+  let ethClient;
+  let subClient;
+
+  before(async function () {
+    const clients = await bootstrap();
+    ethClient = clients.ethClient;
+    subClient = clients.subClient;
+    ethOwner = ethClient.accounts[0];
+    ethUser = ethClient.accounts[1];
+  });
+
+  describe('ERC721 App', function () {
+
+    it('should transfer ERC721 token from Ethereum to Substrate', async function () {
+      const subTokenIds = await subClient.subscribeNFTTokenIdByERC721Id(ethTokenContract, ethTokenId, 2);
+
+      await ethClient.mintERC721(ethTokenId, ethUser, ethOwner);
+
+      const beforeEthOwnerOf = await ethClient.getErc721OwnerOf(ethTokenId);
+      expect(beforeEthOwnerOf).to.be.equal(ethUser);
+      const beforeSubTokenId = await subTokenIds[0]
+      expect(beforeSubTokenId.isNone).to.be.true;
+
+      await ethClient.approveERC721(ethTokenId, ethUser);
+      await ethClient.lockERC721(ethTokenId, ethUser, polkadotRecipient, ChannelId.BASIC);
+
+      const afterEthOwnerOf = await ethClient.getErc721OwnerOf(ethTokenId)
+      expect(afterEthOwnerOf).to.be.equal(ethClient.ERC721AppAddress);
+      const afterSubTokenId = await subTokenIds[1];
+      expect(afterSubTokenId.isSome).to.be.true;
+
+      // TODO why do we need to wait here before the queryNFT command below actually returns the NFT?
```

Comment:
idk - maybe we dont? or i think maybe there was some issue with substrate not actually updating the state/querying the NFT until the block is fully finalized and so wasn't working without waiting an extra block?

anyway, can try fix or just remove this TODO as it's not that important for now

Reply:
Test will fail without waiting – maybe the subscription fires before finality is reached? Not sure and not that relevant I guess – removed the TODO for now.